### PR TITLE
[FIX] mrp: dont mix operations on backorder creation

### PR DESCRIPTION
**Pull request by lase-odoo**
### Steps to reproduce:

1. Create a BOM with 2 operations: op 1 and 2
2. Create and confirm an MO for 10 units using that BOM.
3. Put the quantity to 10 then start and done op 1
4. Put the quantity to 2 then start and done op 2
5. Produce and create a backorder
6. Put the quantity to 4 then start and done op 2
7. Produce and create a backorder
### Cause of the issue:

The operations of a productions (`mrp.workorder`'s) are ordered by `leave_id`, `date_start` and then `id`:
https://github.com/odoo/odoo/blob/26239b2d0bdbc2f06d50f7739d61c490248b65bb/addons/mrp/models/mrp_workorder.py#L14-L17 Since, in the first backorder, the workorder corresponding to op 1 is cancelled it has an empty `leave_id` and `date_start`. On the other hand, once the partial operation of op 2 has been completed its associated `date_start` is not empty anymore and the order of both operations is therefore swaped. However, when a backorder is created, the quantity of the workorders is set by relying on the order of operations (which is not the same with and without `leave_id`, `date_start`):
https://github.com/odoo/odoo/blob/d826c3782d737fc1724887692b71b27999bf846d/addons/mrp/models/mrp_production.py#L1973-L1983 So that the data is set on the wrong line.

opw-3880963
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -1970,7 +1970,7 @@ def _default_amounts(production):
                 workorder.duration_expected = workorder._get_duration_expected()
 
             # Adapt quantities produced
-            for workorder in production.workorder_ids:
+            for workorder in production.workorder_ids.sorted(lambda wo: wo.id):
```
Comment (nle-odoo) [inline addons/mrp/models/mrp_production.py:1973]:
```suggestion
            for workorder in production.workorder_ids.sorted('id'):
```

**Review by nle-odoo — APPROVED** [review]
looks good to me

File: addons/mrp/models/mrp_production.py [hunk 2]
```diff
@@ -1970,7 +1970,7 @@ def _default_amounts(production):
                 workorder.duration_expected = workorder._get_duration_expected()
 
             # Adapt quantities produced
-            for workorder in production.workorder_ids:
+            for workorder in production.workorder_ids.sorted('id'):
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py:1973]: Not sure about this. If I manually change the start date of workorder on the original MO, the order will change, Sorting by `id` here will make the old workorders mismatch the new ones

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**